toolkit/tools/imagepkgfetcher/imagepkgfetcher.go

```diff
@@ -48,105 +40,32 @@ var (
-	} else {
-		err = cloneSystemConfigs(cloner, *configFile, *baseDirPath, *externalOnly, *inputGraph)
-	}
-
-	if err != nil {
-		logger.Log.Panicf("Failed to clone RPM repo. Error: %s", err)
-	}
-
-	logger.Log.Info("Configuring downloaded RPMs as a local repository")
-	err = cloner.ConvertDownloadedPackagesIntoRepo()
-	if err != nil {
-		logger.Log.Panicf("Failed to convert downloaded RPMs into a repo. Error: %s", err)
-	}
-
-	if strings.TrimSpace(*outputSummaryFile) != "" {
-		err = repoutils.SaveClonedRepoContents(cloner, *outputSummaryFile)
-		logger.PanicOnError(err, "Failed to save cloned repo contents")
-	}
-}
-
-func cloneSystemConfigs(cloner repocloner.RepoCloner, configFile, baseDirPath string, externalOnly bool, inputGraph string) (err error) {
-	const cloneDeps = true
-
-	cfg, err := configuration.LoadWithAbsolutePaths(configFile, baseDirPath)
-	if err != nil {
-		return
-	}
-
-	packageVersionsInConfig, err := installutils.PackageNamesFromConfig(cfg)
-	if err != nil {
-		return
-	}
-
-	// Add kernel packages from KernelOptions
-	packageVersionsInConfig = append(packageVersionsInConfig, installutils.KernelPackages(cfg)...)
-
-	if externalOnly {
-		packageVersionsInConfig, err = filterExternalPackagesOnly(packageVersionsInConfig, inputGraph)
-		if err != nil {
-			return
-		}
-	}
-
-	// Add any packages required by the install tools
-	packageVersionsInConfig = append(packageVersionsInConfig, installutils.GetRequiredPackagesForInstall()...)
-
-	logger.Log.Infof("Cloning: %v", packageVersionsInConfig)
-	// The image tools don't care if a package was created locally or not, just that it exists. Disregard if it is prebuilt or not.
-	_, err = cloner.Clone(cloneDeps, packageVersionsInConfig...)
-	return
-}
-
-// filterExternalPackagesOnly returns the subset of packageVersionsInConfig that only contains external packages.
-func filterExternalPackagesOnly(packageVersionsInConfig []*pkgjson.PackageVer, inputGraph string) (filteredPackages []*pkgjson.PackageVer, err error) {
-	dependencyGraph := pkggraph.NewPkgGraph()
-	err = pkggraph.ReadDOTGraphFile(dependencyGraph, inputGraph)
-	if err != nil {
-		return
-	}
-
-	for _, pkgVer := range packageVersionsInConfig {
-		pkgNode, _ := dependencyGraph.FindBestPkgNode(pkgVer)
-
-		// There are two ways an external package will be represented by pkgNode.
-		// 1) pkgNode may be nil. This is possible if the package is never consumed during the build phase,
-		//    which means it will not be in the graph.
-		// 2) pkgNode will be of 'StateUnresolved'. This will be the case if a local package has it listed as
-		//    a Requires or BuildRequires.
-		if pkgNode == nil || pkgNode.RunNode.State == pkggraph.StateUnresolved {
-			filteredPackages = append(filteredPackages, pkgVer)
-		}
-	}
-
-	return
+	cfg := populateImagePkgFetcherConfig()
+	err := pkgfetcher.FetchPkgsAndCreateRepo(cfg)
+	logger.PanicOnError(err, "Failed to save cloned repo contents")
```

Comment:
This looks incorrect now since `pkgfetcher.FetchPkgsAndCreateRepo` can fail for multiple different reasons.